apps/sms/views/shared/js/navigation.js

```diff
@@ -491,28 +491,21 @@ function executeNavigationStep(stepName) {
 }
 
 /**
- * Attach a handler to run the `afterEnter` step after the page is loaded.
- * It uses either the custom event `navigation-transition-end` or the usual
- * `load` event.
- * @returns {Promise} Resolved once afterEnter is run.
+ * Returns promise that is resolved either once Navigation object is marked as
+ * ready (non split views) or when custom `navigation-transition-end` event is
+ * fired (split views).
+ * @returns {Promise.<void>}
  */
-function attachAfterEnterHandler() {
-  if (document.readyState === 'complete') {
-    return executeNavigationStep('afterEnter').catch(catchStepError);
-  }
-
-  var defer = Utils.Promise.defer();
+function waitForThePanelToBeReady() {
+  var readyPromise = readyDefer && readyDefer.promise || Promise.resolve();
 
-  function onNavigationEnd() {
-    window.removeEventListener('navigation-transition-end', onNavigationEnd);
-    window.removeEventListener('load', onNavigationEnd);
-    defer.resolve(executeNavigationStep('afterEnter'));
+  if (isUsingOldArchitecture()) {
```

Comment:
> But we early reject for some cases in "init" :) and in startNavigationFromCurrentLocation too.

Yep, that what I was missing, thanks! :)

> I'm quite sure we can call "toPanel" without init being finshed (using the general case). Actually Navigation's init does a lot more things than the other "init" functions, maybe it's badly named.

Yep, and if init is not rejected, toPanel will wait until navigationTransition promise started by init is finished.